shared/src/bigint/bigint_ser.rs

```diff
@@ -53,5 +59,58 @@ where
             ));
         }
     };
+
+    if bz.len() > MAX_BIGINT_SIZE {
+        return Err(<D::Error as serde::de::Error>::custom("BigInt too large"));
+    }
+
     Ok(BigInt::from_bytes_be(sign, &bz[1..]))
 }
+
+#[cfg(test)]
+mod tests {
+    use fvm_ipld_encoding::{from_slice, to_vec};
+
+    use super::*;
+
+    #[test]
+    fn test_bigiint_max() {
+        let max_limbs = MAX_BIGINT_SIZE / 4; // u32 -> u8
+        let good = BigInt::new(Sign::Plus, vec![u32::MAX; max_limbs - 1]);
+        let good_neg = BigInt::new(Sign::Minus, vec![u32::MAX; max_limbs - 1]);
+
+        let good_bytes = to_vec(&BigIntSer(&good)).expect("should be good");
+        let good_back: BigIntDe = from_slice(&good_bytes).unwrap();
+        assert_eq!(good_back.0, good);
+
+        let good_neg_bytes = to_vec(&BigIntSer(&good_neg)).expect("should be good");
+        let good_neg_back: BigIntDe = from_slice(&good_neg_bytes).unwrap();
+        assert_eq!(good_neg_back.0, good_neg);
+
+        let bad1 = BigInt::new(Sign::Plus, vec![u32::MAX; max_limbs]);
```

Comment:
It took me a while to understand why this is a bad `BigInt`. So perhaps it would be good to add a comment that the sign adds an additional byte when serialized (feel free to ignore if you think that's only me).

Reply:
added a comment